### DISCUSSION TITLE
OLED brightness and Gamma table manipulation

### DISCUSSION THREAD

**Discussion by samleemc** [MonitorControl/MonitorControl#1367] [General]
### Before opening the issue, have you...?

- [X] Searched for existing issues

### Question

For OLED, in the readme, it mentioned that

> OLED or mini/micro-LED displays and televisions are fully supported using gamma table manipulation (this is a no-compromise solution for this class of displays).

Would someone please share some light on the following questions:
1. Will MonitorControl automatically determine whether the connected monitor is an OLED/Mini LED and use Gamma table manipulation for brightness control?
2. (More in the depth of the topic) why controlling the brightness using gamma table manipulation on OLED/Mini LED is no-compromise? Compare to normal LCD monitor? (I tried to search on Google but couldn't find any relevant topic)

**Comment by waydabber** [comment]
Hi there,

1. No, you need to disable hardware brightness control so the app will fall back to gamma table manipulation
2. OLED displays don't have a backlight so hardware backlight control does not make much sense. With mini-LED the best approach is a combination of hardware and gamma control (if supported) as even though mini led displays can scale the mini-led zones to full black using software-only, the algorithm will create more blooming. Micro-LED displays (like OLEDs) should not have this issue (although micro-LED displays are very rare, Samsung has some super-expensive expandable micro-LED TVs).
